dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx

```diff
@@ -0,0 +1,1701 @@
+      setCustomCA(repo.tlsConfig?.certAuthority || "");
+      setAuthCustomHeader(repo.auth?.header || "");
+      setBearerToken(repo.auth?.header || "");
+      setBasicPassword(repo.auth?.usernamePassword?.password || "");
+      setBasicUser(repo.auth?.usernamePassword?.username || "");
+      setSecretEmail(repo.auth?.dockerCreds?.email || "");
+      setSecretPassword(repo.auth?.dockerCreds?.password || "");
+      setSecretServer(repo.auth?.dockerCreds?.server || "");
+      setSecretUser(repo.auth?.dockerCreds?.username || "");
+      setSshKnownHosts(repo.auth?.sshCreds?.knownHosts || "");
+      setSshPrivateKey(repo.auth?.sshCreds?.privateKey || "");
+      setTlsAuthCert(repo.auth?.tlsCertKey?.cert || "");
+      setTlsAuthKey(repo.auth?.tlsCertKey?.key || "");
+      setOpaqueData(JSON.stringify(repo.auth?.opaqueCreds?.data) || "");
+      setAuthMethod(
+        repo.auth?.type ||
+          PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_UNSPECIFIED,
+      );
+      setSecretAuthName(repo.auth?.secretRef?.name || "");
+      setSecretTLSName(repo.tlsConfig?.secretRef?.name || "");
+      setIsUserManagedSecret(!!repo.auth?.secretRef?.name);
+
+      // setting custom details for the Helm plugin
+      if (repo.packageRepoRef?.plugin?.name === PluginNames.PACKAGES_HELM) {
+        const repositoryCustomDetails = repo.customDetail as Partial<RepositoryCustomDetails>;
+        setOCIRepositories(repositoryCustomDetails?.ociRepositories?.join(", ") || "");
+        setPerformValidation(repositoryCustomDetails?.performValidation || false);
+        if (repositoryCustomDetails?.filterRule?.jq) {
+          const { names, regex, exclude } = toParams(repositoryCustomDetails.filterRule!);
+          setFilterRegex(regex);
+          setFilterExclude(exclude);
+          setFilterNames(names);
+        }
+        setHelmPsAuthMethod(
+          repositoryCustomDetails?.dockerRegistrySecrets?.length
+            ? PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
+            : PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_UNSPECIFIED,
+        );
+      }
+    }
+  }, [repo, namespace, currentCluster, dispatch]);
+
+  const handleInstallClick = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    install();
+  };
+
+  const install = async () => {
+    if (isInstallingRef.current) {
+      // Another installation is ongoing
+      return;
+    }
+    isInstallingRef.current = true;
+
+    // send the proper header depending on the auth method
+    let finalHeader = "";
+    switch (authMethod) {
+      case PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_AUTHORIZATION_HEADER:
+        finalHeader = authCustomHeader;
+        break;
+      case PackageRepositoryAuth_PackageRepositoryAuthType.PACKAGE_REPOSITORY_AUTH_TYPE_BEARER:
+        finalHeader = `Bearer ${bearerToken}`;
```

Comment:
Good catch! I've added to the main issue (https://github.com/vmware-tanzu/kubeapps/issues/4764) as a blocker thing.
I have mixed opinions on this problem: on the one hand, I don't see why the API should care about something else than an "HTTP header authentication" with two alternatives: either an RFC-compliant _Authorization_ header or a custom header.
However, on the other hand, we would be delegating that logic to the clients, but... why not? 

I'll post it in the issue as well, so that we can continue the discussion there.